feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/encap_gre_ipv4_test.go

```diff
@@ -0,0 +1,1062 @@
+		return `
+        qos rewrite dscp
+        !
+        `
+		// cliConfig := fmt.Sprintf("qos map dscp-classifier %s\n", classifierName)
+		// for i, dscp := range ipv4DscpValues {
+		//  cliConfig += fmt.Sprintf(`
+		//  dscp %d class termV4-%d
+		//  qos map dscp-remark remarkV4-%d
+		//  match dscp %d
+		//  set dscp %d
+		//  !
+		//  `, dscp, i+1, i+1, dscp, dscp)
+		// }
+
+		// // Add terms for IPv6 DSCP values
+		// for i, dscp := range ipv6DscpValues {
+		//  cliConfig += fmt.Sprintf(`
+		//  dscp %d class termV6-%d
+		//  qos map dscp-remark remarkV6-%d
+		//  match dscp %d
+		//  set dscp %d
+		//  !
+		//  `, dscp, i+1, i+1, dscp, dscp)
+		// }
+
+		// // Apply the classifier to the interface
+		// cliConfig += fmt.Sprintf(`
+		// interface %s
+		// service-policy input %s
+		// !
+		// `, interfaceName, classifierName)
+
+		//return cliConfig
+	default:
+		return ""
+	}
+}
+
+func trafficPolicyCliConfig(dut *ondatra.DUTDevice, policyName string, interfaceName string) string {
+	switch dut.Vendor() {
+	case ondatra.ARISTA:
+		var v4MatchRules, v6MatchRules string
+		v4MatchRules += fmt.Sprintf(`
+        match rule-src1-v4 ipv4
+        destination prefix %s/32
+        actions
+        count
+        redirect next-hop group SRC1_NH
+        !
+        `, otgPort2.IPv4)
+
+		v4MatchRules += fmt.Sprintf(`
+        match rule-src2-v4 ipv4
+        destination prefix %s/32
+        actions
+        count
+        !
+        `, otgPort3.IPv4)
+
+		v6MatchRules += fmt.Sprintf(`
+        match rule-src1-v6 ipv6
+        destination prefix %s/128
+        actions
+        count
+        redirect next-hop group SRC1_NH
+        !
+        `, otgPort2.IPv6)
+
+		v6MatchRules += fmt.Sprintf(`
+        match rule-src2-v6 ipv6
+        destination prefix %s/128
+        actions
+        count
+        !
+        `, otgPort3.IPv6)
+
+		ipv4GreNH := fmt.Sprintf(`
+        nexthop-group SRC1_NH type gre
+        tunnel-source intf %s
+        `, dutlo0Attrs.Name)
+
+		for index, dest := range tunnelDestinations {
+			ipv4GreNH += fmt.Sprintf(`
+            entry %d tunnel-destination %s
+            `, index, dest)
+		}
+
+		// Apply Policy on the interface
+		trafficPolicyConfig := fmt.Sprintf(`
+            traffic-policies
+            traffic-policy %s
+            %s
+            %s
+            %s
+            !
+            interface %s
+            traffic-policy input %s
+            `, policyName, v4MatchRules, v6MatchRules, ipv4GreNH, interfaceName, trafficPolicyName)
+		return trafficPolicyConfig
+	default:
+		return ""
+	}
+}
+
+func checkPolicyStatistics(t *testing.T, dut *ondatra.DUTDevice, tc testCase) {
+	if deviations.PolicyForwardingGreEncapsulationOcUnsupported(dut) {
+		checkPolicyStatisticsFromCLI(t, dut, tc)
+	} else {
+		checkPolicyStatisticsFromOC(t, dut, tc)
+	}
+}
+
+func checkPolicyStatisticsFromCLI(t *testing.T, dut *ondatra.DUTDevice, tc testCase) {
+	t.Logf("Checking policy statistics for flow %s", tc.flowName)
```

Comment:
Per the [deviations guide, we cannot accept any deviations for telemetry using CLI](https://github.com/openconfig/featureprofiles/blob/main/internal/deviations/README.md).    The test must use OC or vendor yang paths for telemetry.    If no gnmi streaming option for telemetry exists on the DUT, then the test should fail.